### PR TITLE
SVGO optimizer

### LINKED ISSUE #6365: @parcel/transformer-svg-react "Namespace tags are not supported by default" error
While diagnosing #6362 (importing the `logo.svg` file attached to that issue), I ran into the "Namespace tags are not supported by default" error mentioned in gregberge/svgr#271.

If I run the svgr CLI and generate a React component manually, via `npx @svgr/cli logo.svg > Logo.jsx`, I'm able to use the resulting module without issue.  But if I do `import Logo from './logo.svg'`, I get the error in the title. 

What's the best way to resolve this?  The error message suggests setting `throwIfNamespace: false`.  But how/where do I do that?    (Is this a .babelrc thing?  If so, how do I set that up without being taken down the path of having to install a bunch of parcel and babel plugins and presets just to get back to the current baseline functionality Parcel provides?

[Edit to add...]
## envinfo
```
  System:
    OS: macOS 11.1
    CPU: (8) x64 Apple M1
    Memory: 38.84 MB / 16.00 GB
    Shell: 5.1.0 - /usr/local/bin/bash
  Binaries:
    Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
    npm: 6.14.12 - ~/.nvm/versions/node/v14.16.1/bin/npm
  Browsers:
    Chrome: 91.0.4472.77
    Firefox: 88.0.1
    Safari: 14.0.2
  npmPackages:
    @parcel/babel-plugin-transform-runtime: 2.0.0-nightly.1823 => 2.0.0-nightly.1823 
    @parcel/packager-raw-url: 2.0.0-beta.2 => 2.0.0-beta.2 
    @parcel/transformer-sass: 2.0.0-beta.2 => 2.0.0-beta.2 
    @parcel/transformer-svg-react: 2.0.0-beta.3.1 => 2.0.0-beta.3.1 
    @parcel/transformer-webmanifest: 2.0.0-beta.2 => 2.0.0-beta.2 
    @types/react-dom: 17.0.3 => 17.0.3 
    @types/react-router-dom: 5.1.7 => 5.1.7 
    @typescript-eslint/eslint-plugin: 4.20.0 => 4.20.0 
    @typescript-eslint/parser: 4.20.0 => 4.20.0 
    bootstrap: 5.0.1 => 5.0.1 
    checkcss: 1.0.2 => 1.0.2 
    eslint: 7.21.0 => 7.21.0 
    eslint-config-standard: 16.0.2 => 16.0.2 
    eslint-plugin-import: 2.22.1 => 2.22.1 
    eslint-plugin-node: 11.1.0 => 11.1.0 
    eslint-plugin-promise: 4.3.1 => 4.3.1 
    eslint-plugin-react: 7.22.0 => 7.22.0 
    eslint-plugin-react-hooks: 4.2.0 => 4.2.0 
    firebase: 8.4.2 => 8.4.2 
    firebaseui: 4.8.0 => 4.8.0 
    nanoid: 3.1.22 => 3.1.22 
    parcel: 2.0.0-beta.2 => 2.0.0-beta.2 
    react: ^16.13.1 => 16.14.0 
    react-bootstrap: 2.0.0-alpha.2 => 2.0.0-alpha.2 
    react-dom: ^16.13.1 => 16.14.0 
    react-router-bootstrap: 0.25.0 => 0.25.0 
    react-router-dom: 5.2.0 => 5.2.0 
    sass: 1.32.12 => 1.32.12 
    simplur: 3.0.0 => 3.0.0 
    typescript: 4.2.3 => 4.2.3 
```

### PR DESCRIPTION
**Pull request by mischnic**
Removes the SVGO transformer (by moving SVGO into the react-svg transformer) and adds a SVGO optimizer

Closes T-1080
Closes https://github.com/parcel-bundler/parcel/issues/6365

One problem with svgo 2 is that it's really designed for JS config files. The way to disable only one plugin and keep the rest at the default setting is
```js
// svgo.config.js
const { extendDefaultPlugins } = require('svgo');
module.exports = {
  plugins: extendDefaultPlugins([
    {
      name: 'builtinPluginName',
      active: false
    }
  ])
}
```
https://github.com/svg/svgo#configuration

https://github.com/svg/svgo/issues/1497

Not entirely sure if SVGO in the react-svg transformer should only be enabled if `shouldOptimize`, because svg files from editors can be massive

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.26s | -185.00ms |
| Cached | 467.00ms | -11.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.1e014c76.png | 274.00b | +0.00b | 4.94s | +4.64s ⚠️ |

</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.41s | +1.00ms |
| Cached | 416.00ms | +5.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/302538799957672457)**

File: packages/optimizers/svgo/src/SVGOOptimizer.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export default (new Optimizer({
+  async loadConfig({config}) {
+    let configFile = await config.getConfig(
+      [
+        '.svgorc',
+        '.svgorc.json',
+        '.svgorc.yaml',
+        '.svgorc.yml',
+        'svgo.config.js',
+        '.svgo.yml',
```
Comment (mischnic) [inline packages/optimizers/svgo/src/SVGOOptimizer.js]: The only config files read by svgo 2 itself is `svgo.config.js`, so maybe we should only support `svgo.config.js(on)`

**Review by devongovett — APPROVED** [review]
Looks good. I agree re only supporting the config filenames that SVGO supports officially.

**Review by devongovett — APPROVED** [review]

File: packages/optimizers/svgo/src/SVGOOptimizer.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import path from 'path';
+
+export default (new Optimizer({
+  async loadConfig({config}) {
+    let configFile = await config.getConfig([
+      'svgo.config.js',
+      'svgo.config.json',
+    ]);
+
+    if (configFile) {
```
Comment (steinwaywhw) [inline packages/optimizers/svgo/src/SVGOOptimizer.js:17]: What happens when `if (configFile)` evaluates to `false`? The function does not handle this case and it triggered this line: https://github.com/svg/svgo/blob/77102ed0963a5f1597e8072536311b55465e1874/lib/svgo.js#L21

I encountered this issue on parcel 2.0.0-rc.0. I have SVG, but I don't have a corresponding SVG Optimizer config. I do not have .parcelrc overrides.

Comment (thewilkybarkid) [reply packages/optimizers/svgo/src/SVGOOptimizer.js:17]: I've opened https://github.com/svg/svgo/pull/1573. Sounds like you could be seeing what I've just found, which is actually https://github.com/svg/svgo/blob/721a1e591da131affece29362cf7e7089b474904/lib/svgo-node.js#L57-L59 (which doesn't have the null-ish check).

**Pull request merged**